ts/src/hashkey.ts

```diff
@@ -0,0 +1,4336 @@
+        //                 "initialMargin": "0.10",
+        //                 "maintMargin": "0.005",
+        //                 "isWhite": false
+        //             },
+        //             {
+        //                 "riskLimitId": "200000723",
+        //                 "quantity": "2000.00",
+        //                 "initialMargin": "0.10",
+        //                 "maintMargin": "0.01",
+        //                 "isWhite": false
+        //             }
+        //         ]
+        //     }
+        //
+        const marketId = this.safeString (market, 'symbol');
+        const quoteId = this.safeString (market, 'quoteAsset');
+        const quote = this.safeCurrencyCode (quoteId);
+        const settleId = this.safeString (market, 'marginToken');
+        const settle = this.safeCurrencyCode (settleId);
+        let baseId = this.safeString (market, 'baseAsset');
+        let marketType = 'spot' as any;
+        let isSpot = true;
+        let isSwap = false;
+        let suffix = '';
+        const parts = marketId.split ('-');
+        const secondPart = this.safeString (parts, 1);
+        if (secondPart === 'PERPETUAL') {
+            marketType = 'swap';
+            isSpot = false;
+            isSwap = true;
+            baseId = this.safeString (market, 'underlying');
+            suffix += ':' + settleId;
+        }
+        const base = this.safeCurrencyCode (baseId);
+        const symbol = base + '/' + quote + suffix;
+        const status = this.safeString (market, 'status');
+        const active = status === 'TRADING';
+        let isLinear: Bool = undefined;
+        let subType = undefined;
+        const isInverse = this.safeBool (market, 'inverse');
+        if (isInverse !== undefined) {
+            if (isInverse) {
+                isLinear = false;
+                subType = 'inverse';
+            } else {
+                isLinear = true;
+                subType = 'linear';
+            }
+        }
+        const filtersList = this.safeList (market, 'filters', []);
+        const filters = this.indexBy (filtersList, 'filterType');
+        const priceFilter = this.safeDict (filters, 'PRICE_FILTER', {});
+        const amountFilter = this.safeDict (filters, 'LOT_SIZE', {});
+        const costFilter = this.safeDict (filters, 'MIN_NOTIONAL', {});
+        const minCostString = this.omitZero (this.safeString (costFilter, 'min_notional'));
+        const contractSizeString = this.safeString (market, 'contractMultiplier');
+        let amountPrecisionString = this.safeString (amountFilter, 'stepSize');
+        let amountMinLimitString = this.safeString (amountFilter, 'minQty');
+        let amountMaxLimitString = this.safeString (amountFilter, 'maxQty');
+        let minLeverage: Int = undefined;
+        let maxLeverage: Int = undefined;
+        if (isSwap) {
+            amountPrecisionString = Precise.stringDiv (amountPrecisionString, contractSizeString);
```

Comment:
I think this is wrong, it seems that `stepSize` is the amount precision already you don't need to divide here

Reply:
The documentation of the exchange says that all swap amount of contracts should be an Int. Deleting this division causes that `amountToPresicion` doesn't throw an exception when amount is less than 1, and the exchange returns `{"code":"-1202","msg":"Order sell quantity too small"}`